shared/protocol/src/test/java/io/pravega/shared/protocol/netty/AppendEncodeDecodeTest.java

```diff
@@ -219,23 +219,37 @@ public void execute(Runnable runnable) {
 
         }
     };
-
     @Mock
     private ChannelHandlerContext ctx;
     @Mock
     private Channel ch;
+    private ByteBuf fakeNetwork;
+    private BufferReleaseVerifier releaseVerifier;
+
+    protected int getThreadPoolSize() {
+        return 1;
```

Comment:
Does this number have to be 1 for the test to pass? If so there should be a comment explaining it.

Reply:
If it's 0, it uses the InlineExecutor. If it's 1, it uses an executor with 1 thread. This is exactly as it was before, if you look at the modifications in this class, except that now I am using the services from the inherited class as opposed from doing it again.